File: GO/functions/decimal_to_binary.go
package main
import "fmt"

func decimal_to_binary (num int) int {
	res := 0
	level := 1
	for (num > 0) {
		var tempI = num / 2
		var tempF = num*1.0 / 2
		fmt.Printf("tempI is %f\n", tempI)
		fmt.Printf("tempF is %f\n", tempF)
		res += tempF * level
		level *= 10
		num = tempI
	}
	return res
}

func main () {
	var x int
	var results int
	fmt.Println("User, enter an INTEGER: ")
	fmt.Scan(&x)
	results = decimal_to_binary(x)
	fmt.Printf("%d --> %d", x, results)
}

File: PYTHON/Pure Functions/decimal to binary.py
def decimal_to_binary (num):
    num = int(num)
    res = int(0)
    level = int(1)
    while (num > 0):
        tempI = int(num / 2)
        tempF = float(num*1.0 / 2)
        tempF = float(tempF - tempI)
        tempF = int(tempF * 2)
        res = int(res + level * tempF)
        level = int(level * 10)
        num = int(tempI)
    return res

x = input("User, enter a number:\n")
x = int(x)

results = decimal_to_binary(x)
print("The results are: " + str(results))
input("\nDONE!\n")